devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java

```diff
@@ -262,26 +275,49 @@ private QuarkusDevModeLauncher newLauncher() throws Exception {
 
         final ApplicationModel appModel = extension().getApplicationModel(LaunchMode.DEVELOPMENT);
         final Set<GACT> projectDependencies = new HashSet<>();
+
         addSelfWithLocalDeps(project, builder, new HashSet<>(), projectDependencies, true);
 
+        //look for an application.properties
+        Set<Path> resourceDirs = new HashSet<>();
+        for (SourceDir resourceDir : appModel.getApplicationModule().getMainSources().getResourceDirs()) {
+            resourceDirs.add(resourceDir.getOutputDir());
+        }
+
+        Set<ArtifactKey> configuredParentFirst = QuarkusBootstrap.createClassLoadingConfig(PathsCollection.from(resourceDirs),
+                QuarkusBootstrap.Mode.DEV, Collections.emptyList()).parentFirstArtifacts;
+
+        Set<ArtifactKey> parentFirstArtifactKeys = new HashSet<>(configuredParentFirst);
+        for (io.quarkus.maven.dependency.ResolvedDependency resolvedDependency : appModel.getDependencies()) {
```

Comment:
Now this change based on the ApplicationModel API made me realize we can unify this handling across build systems.
I won't ask you to do this @glefloch unless you really want to.
A few remarks though. You can check `resolvedDependency.isRuntimeCp()` before opening its JAR to filter out the deployment deps.
Second, you could use
````
resolvedDependency.getContentTree().apply(BootstrapConstants.DESCRIPTOR_PATH, visit -> {
    if(visit == null)
        return null;
    // read and return the properties from visit.getPath()
});
````

Reply:
Actually, even better would be `resolvedDependency.isRuntimeExtensionArtifact()` that way we already know it contains the descriptor and will avoid opening other JARs for nothing.

Reply:
Actually. there is `Set<ArtifactKey> ApplicationModel.getParentFirst()`

Reply:
I will use that set, both here and  in the maven plugin, this will simplify things.